Lib/selectors.py

```diff
@@ -294,43 +298,52 @@ def __init__(self):
         super().__init__()
         self._readers = set()
         self._writers = set()
+        self._urgents = set()
 
     def register(self, fileobj, events, data=None):
         key = super().register(fileobj, events, data)
         if events & EVENT_READ:
             self._readers.add(key.fd)
         if events & EVENT_WRITE:
             self._writers.add(key.fd)
+        if events & EVENT_URGENT:
+            self._urgents.add(key.fd)
         return key
 
     def unregister(self, fileobj):
         key = super().unregister(fileobj)
         self._readers.discard(key.fd)
         self._writers.discard(key.fd)
+        self._urgents.discard(key.fd)
         return key
 
     if sys.platform == 'win32':
-        def _select(self, r, w, _, timeout=None):
-            r, w, x = select.select(r, w, w, timeout)
-            return r, w + x, []
+        def _select(self, r, w, x, timeout=None):
+            r, _w, _x = select.select(r, w, w|x, timeout)
```

Comment:
It would be nice to add a comment explaining the `w|x` thing as well as the `_w+_x` below.